### PR TITLE
Changelogger: Consolidate JS packages' use of composer into one

### LINKED ISSUE #38989: Changelogger: Consolidate JS packages' use of composer into one
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

Currently each JS package has a composer.json file to include PHP dependencies of which there is only one - Jetpack Changelogger. This causes a delay in `pnpm install` and requires an extra `pnpm install` (which postinstalls `composer`) in some CLI actions. A solution would be to have one top level instance of Changelogger.

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/38989

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. `pnpm install`
2. Check changelog CLI functions as it normally does
```
pnpm --filter=woocommerce run changelog add
pnpm --filter=@woocommerce/currency run changelog add
```
3. See the `Add changelog to PR` action pass and resulting changefiles with every touched project

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement

#### Message 

Consolidate Jetpack Changelogger dependency into single tool.

#### Comment 

</details>

### CODE REVIEW

File: tools/monorepo/check-changelogger-use.php [hunk 1]
```diff
@@ -112,16 +115,15 @@ function debug() {
...
+			! isset( $data['extra']['changelogger'] )
 		) {
 			continue;
 		}
 	} catch ( Exception $e ) {
 		continue;
 	}
-	$data  = isset( $data['extra']['changelogger'] ) ? $data['extra']['changelogger'] : array();
-	$data += array(
+	$data                                   = isset( $data['extra']['changelogger'] ) ? $data['extra']['changelogger'] : array();
```
Comment (jonathansadowski) [inline tools/monorepo/check-changelogger-use.php:125]: Minor suggestion: if we're continuing above when `$data['extra']['changelogger']` isn't set, we shouldn't need this `isset` check here, correct?

```suggestion
	$data                                   = $data['extra']['changelogger'];
```

**Review by jonathansadowski — CHANGES_REQUESTED** [review]
I added a minor suggestion about some potentially unneeded code after the removal of the check for `automattic/jetpack-changelogger` in the `require` or `require-dev` section, but that also brought up a question:

Do we need to add `.extra.changelogger` to the `composer.json` files for each project that we want checked, or is there some other way that I'm missing that we're indicating that a project requires a change file?

**Comment by psealock** [comment]
>Do we need to add .extra.changelogger to the composer.json files for each project that we want checked, or is there some other way that I'm missing that we're indicating that a project requires a change file?

Good call out here. Indeed the current way of checking `.extra.changelogger` won't work if we ever have a package or plugin that simply uses the default settings of Jetpack Changelogger. As it is now all our packages/project have this property and new ones would require it. That is less than ideal. Some projects are internal and don't require changelogs while others don't use Changelogger such as the e2e packages. Can you think of a good way to differentiate now that individual projects don't require Jetpack Changelogger?

**Comment by jonathansadowski** [comment]
> Can you think of a good way to differentiate now that individual projects don't require Jetpack Changelogger?

Maybe it would be easier to just maintain a config of projects that require changelogger, and if it's not in that list, it won't check. A bit of a pain to maintain, but more straight-forward.

**Comment by psealock** [comment]
Using `time pnpm install` on this branch and `trunk` I'm seeing a ~2 second reduction in time, which isn't too impressive.

@jonathansadowski, I've updated the mechanism for distinguishing Jetpack Changelogger projects to use a config option in `composer.json`:

```json
"config": {
	"changelog": "true"
}
```

**Comment by psealock** [comment]
ping @jonathansadowski if you have a chance to re-review 🙇🏽‍♂️

**Pull request closed without merging**